router/core/router.go

```diff
@@ -1050,6 +1106,12 @@ func (r *Router) Shutdown(ctx context.Context) (err error) {
 		ctx = ctxWithTimer
 	}
 
+	if r.configFileWatcher != nil {
+		if subErr := r.configFileWatcher.Close(); subErr != nil {
```

Comment:
Can a return happen before this?
If yes, I'd implement this cleanup with a defer after creating the watcher didn't return an error.
Otherwise ignore me.

Reply:
No it can't. All cleanup functions never return and the shutdown function will return joined errors.